packages/test/test-end-to-end-tests/src/test/gc/gcTrailingOps.spec.ts

```diff
@@ -17,200 +17,230 @@ import {
+			);
+			assert.equal(
+				dataStoreGCData.unreferencedTimestampMs ? false : true,
+				referenced,
+				`Data store ${dataStoreNodePath}'s referenced state is incorrect`,
+			);
+
+			// Validate that the deleted nodes in the GC data does not have the data store.
+			const deletedNodesState = getGCDeletedStateFromSummary(summaryTree);
+			assert.equal(
+				deletedNodesState?.includes(dataStoreNodePath) ?? false,
+				false,
+				`Data store ${dataStoreNodePath} should not be in deleted nodes`,
+			);
 		}
 
 		beforeEach(async function () {
 			provider = getTestObjectProvider({ syncSummarizer: true });
+			if (provider.driver.type !== "local") {
+				this.skip();
+			}
+			settings["Fluid.GarbageCollection.TestOverride.SweepTimeoutMs"] = sweepTimeoutMs;
+
+			const configProvider = mockConfigProvider(settings);
+			testContainerConfig = {
+				runtimeOptions: {
+					summaryOptions: {
+						summaryConfigOverrides: {
+							state: "disabled",
+						},
+					},
+					gcOptions,
+				},
+				loaderProps: { configProvider },
+			};
+
+			summarizerContainerConfig = {
+				runtimeOptions: { gcOptions },
+				loaderProps: { configProvider },
+			};
+		});
+
+		afterEach(() => {
+			settings = {};
 		});
 
-		it(`A summary has a datastore and blob referenced, but trailing ops unreferenced them ${
-			tombstoneEnabled ? "after sweep timeout" : "before sweep timeout"
-		}`, async () => {
+		it(`Trailing ops ${
+			beforeSweepTimeout ? "before sweep timeout" : "after sweep timeout"
+		} makes data store ${
+			transitionToReference ? "referenced" : "unreferenced"
+		} without deleting it`, async () => {
 			const mainContainer = await provider.makeTestContainer(testContainerConfig);
-			const mainDefaultDataStore = (await mainContainer.getEntryPoint()) as ITestDataObject;
+			const mainDataStore = (await mainContainer.getEntryPoint()) as ITestDataObject;
 			await waitForContainerConnection(mainContainer);
 
-			// Create a data store and blob.
+			const newDataStoreKey = "datastore";
+			// Create a data store and reference it.
 			const newDataStore =
-				await mainDefaultDataStore._context.containerRuntime.createDataStore(
-					TestDataObjectType,
-				);
+				await mainDataStore._context.containerRuntime.createDataStore(TestDataObjectType);
 			assert(newDataStore.entryPoint !== undefined, `Should have a handle`);
-			const blobContents = "Blob contents";
-			const blobHandle = await mainDefaultDataStore._runtime.uploadBlob(
-				stringToBuffer(blobContents, "utf-8"),
+			const newTestDataObject = (await newDataStore.entryPoint.get()) as ITestDataObject;
+			mainDataStore._root.set(newDataStoreKey, newDataStore.entryPoint);
+
+			// Update the reference state. The data store should start in the opposite state of "transitionToReference"
+			// and the trailing op will transition it to that state.
+			updateDataStoreReferenceState(
+				mainDataStore,
+				newTestDataObject,
+				newDataStoreKey,
+				!transitionToReference,
 			);
 
```

Comment:
Sometimes I'm searching the code for a test by name and can't find it, I have to start picking random substrings because I don't know where the test name is parametrized.  An idea I just had would be to put the variable bits in `[]`, so the output is like `Trailing ops [after sweep timeout] makes data store [unreferenced] without deleting it`.  It's kind of ugly, so not sure about it, but it would help find the tests if they start failing in some CI run.

Reply:
let's try it :-).